hazelcast/src/main/java/com/hazelcast/spi/impl/operationservice/impl/InvocationMonitor.java

```diff
@@ -47,134 +61,178 @@
-    private static final int SCAN_DELAY_MILLIS = 1000;
+    private static final long ON_MEMBER_LEFT_DELAY_MILLIS = 1111;
+    private static final int HEARTBEAT_CALL_TIMEOUT_RATIO = 4;
 
-    private final long backupTimeoutMillis;
+    private final NodeEngineImpl nodeEngine;
+    private final InternalSerializationService serializationService;
+    private final ServiceManager serviceManager;
     private final InvocationRegistry invocationRegistry;
-    private final ExecutionService executionService;
-    private final InvocationMonitorThread monitorThread;
     private final ILogger logger;
+    private final ScheduledExecutorService scheduler;
     @Probe(name = "backupTimeouts", level = MANDATORY)
     private final SwCounter backupTimeoutsCount = newSwCounter();
     @Probe(name = "normalTimeouts", level = MANDATORY)
     private final SwCounter normalTimeoutsCount = newSwCounter();
-
-    public InvocationMonitor(InvocationRegistry invocationRegistry, ILogger logger, GroupProperties props,
-                             HazelcastThreadGroup hzThreadGroup, ExecutionService executionService,
-                             MetricsRegistry metricsRegistry) {
+    @Probe
+    private final long backupTimeoutMillis;
+    @Probe
+    private final long invocationTimeoutMillis;
+    @Probe
+    private final long slowInvocationThresholdMillis;
+    @Probe
+    private final long heartbeatBroadcastPeriodMillis;
+    @Probe
+    private final long invocationScanPeriodMillis = SECONDS.toMillis(1);
+    @Probe
+    private volatile long heartbeatPacketsReceived;
+    @Probe
+    private volatile long heartbeatPacketsSend;
+
+    public InvocationMonitor(InvocationRegistry invocationRegistry, ILogger logger, NodeEngineImpl nodeEngine) {
+        this.nodeEngine = nodeEngine;
+        this.serializationService = (InternalSerializationService) nodeEngine.getSerializationService();
+        this.serviceManager = nodeEngine.getServiceManager();
         this.invocationRegistry = invocationRegistry;
         this.logger = logger;
-        this.executionService = executionService;
-        this.backupTimeoutMillis = props.getMillis(GroupProperty.OPERATION_BACKUP_TIMEOUT_MILLIS);
-        this.monitorThread = new InvocationMonitorThread(hzThreadGroup);
+        this.backupTimeoutMillis = nodeEngine.getGroupProperties().getMillis(OPERATION_BACKUP_TIMEOUT_MILLIS);
+        this.invocationTimeoutMillis = invocationTimeoutMillis();
+        this.heartbeatBroadcastPeriodMillis = heartbeatBroadcastPeriodMillis();
+        this.slowInvocationThresholdMillis = slowInvocationThresholdMillis();
+
+        this.scheduler = newScheduler();
+
+        nodeEngine.getMetricsRegistry().scanAndRegister(this, "operation.invocations");
+    }
+
+    private ScheduledExecutorService newScheduler() {
+        // the scheduler is configured with a single thread.
+        ScheduledExecutorService scheduler = new ScheduledThreadPoolExecutor(1, new ThreadFactory() {
```

Comment:
for future reference: I was wrong here. This is a contract of `ScheduledThreadPoolExecutor`:
```
 * <p>While this class inherits from {@link ThreadPoolExecutor}, a few
 * of the inherited tuning methods are not useful for it. In
 * particular, because it acts as a fixed-sized pool using
 * {@code corePoolSize} threads and an unbounded queue, adjustments
 * to {@code maximumPoolSize} have no useful effect. Additionally, it
 * is almost never a good idea to set {@code corePoolSize} to zero or
 * use {@code allowCoreThreadTimeOut} because this may leave the pool
 * without threads to handle tasks once they become eligible to run.
```